consensus/core/src/core.rs

```diff
@@ -131,7 +131,12 @@ impl Core {
         self.add_accepted_blocks(last_quorum);
         // Try to commit and propose, since they may not have run after the last storage write.
         self.try_commit().unwrap();
-        self.try_propose(true).unwrap();
+        if self.try_propose(true).unwrap().is_none() {
+            // if no new block proposed then just re-broadcast the last proposed one to ensure liveness.
+            self.signals
+                .new_block(self.last_proposed_block.clone())
```

Comment:
Can you make sure this is not the genesis block?

Reply:
Sure I can, but by design this should always be at minimum the block of round 1. But let me put an assert there because is indeed important